src/app/utils/Encoder.ts

```diff
@@ -7,7 +7,34 @@ export abstract class Encoder<T> {
     public abstract decode(encoded: JSONValue): T;
 }
 
+// Used internally. If T = [A, B, C], then
+// EncoderArray<T> = [Encoder<A>, Encoder<B>, Encoder<C>]
+type EncoderArray<T> = { [P in keyof T]: Encoder<T[P]> };
+
 export abstract class MoveEncoder<T> extends Encoder<T> {
+    public static tuple<T, Fields extends object>(encoders: EncoderArray<Fields>,
```

Comment:
Y avais pas déjà un tuple quelque part ?

Reply:
Ouais il y a un tuple pour encoder les nombres
C'est pas la première apparition d'object, il est dans l'autre tuple notamment, en gros c'est le type de base pour un objet qui contient des champs, c'est la façon recommandée de typer ceci

Reply:
Yep, du typeof pilou === 'object' et l'autre tuple yep.
EN gros c'est un jason quoi ?
On aurais quoi comme choix alternatifs pour typer ça (et ils sont pas la façon recommandé car ?) (si t'as pas ça top of your mind c'pas grave :D )

Reply:
`object` c'est la maman de tout le monde sauf de `null` et `undefined`, je pense !
Ça me semble le choix le plus safe qu'on peut avoir en TypeScript !